src/protocols/gamma_control.rs

```diff
@@ -160,37 +161,24 @@ where
     ) {
         match request {
             zwlr_gamma_control_v1::Request::SetGamma { fd } => {
-                info!("Setting gamma for output {:?}", data.output);
-                let buf = &mut Vec::new();
-                if File::from(fd).read_to_end(buf).is_err() {
-                    error!("Failed to read gamma data");
+                if data.failed {
                     resource.failed();
                     return;
                 }
-
-                let mut gamma = Vec::new();
-                for i in 0..buf.len() / 2 {
-                    gamma.push(((buf[i * 2 + 1] as u16) << 8) | buf[i * 2] as u16);
-                }
-                if data.gamma_size.is_none() {
-                    error!("gamma size not set");
+                debug!("setting gamma for output {:?}", data.output);
+                let buf = &mut Vec::new();
+                if File::from(fd).read_to_end(buf).is_err() {
+                    warn!("failed to read gamma data for output {:?}", data.output);
                     resource.failed();
```

Comment:
Also set `data.failed = true;`

Reply:
data itself is read only here and the Dispatch trait doesn't allow &mut data. We could get arround it with some research on data_init maybe, but I think, setting data.failed is not vital here, what do you think?

Reply:
Ah, right, you'd need Mutexes for that... Well I guess it's not a big deal. The client is supposed to destroy upon failing anyway.